datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1337,12 +1339,33 @@ private void handleLeaderDoAssignment(boolean cleanUpOrphanNodes) {
         _adapter.updateAllAssignments(newAssignmentsByInstance);
       }
 
+      Set<String> failedStreams = Collections.emptySet();
+      // Poll the zookeeper to ensure that hosts claimed assignment tokens for stopping streams
+      if (_config.getEnableAssignmentTokens() &&
+          !PollUtils.poll(() -> _adapter.getNumUnclaimedTokensForDatastreams(stoppingDatastreamGroups) == 0,
+            STOP_PROPAGATION_RETRY_MS, _config.getStopPropagationTimeout())) {
+        Map<String, List<AssignmentToken>> unclaimedTokens =
+            _adapter.getUnclaimedAssignmentTokensForDatastreams(stoppingDatastreamGroups);
```

Comment:
If there is a bug and the list of hosts that need to claim the token is incorrect, how will we debug it?

What happens if the instance dies in between and is not able to claim the token? and then if the task moves to another host?

Reply:
The cleanup code will be in Part 4. I'll see what other logs/metrics I can emit to help us debug issues with hosts dying.